pytket/binders/placement.cpp

```diff
@@ -103,7 +65,7 @@ PYBIND11_MODULE(placement, m) {
             .def("__repr__",
                  [](const Placement &) { return "<tket::Placement>"; })
             .def("place",
-              [](const Placement &placement, Circuit &circ) {
+              [](Placement &placement, Circuit &circ) {
```

Comment:
The `Architecture` held in `Placement` is not guaranteed `const` because they can cache the distance between their `Node` as they are found for the first time. I'm not entirely sure how we got away with it being const before, but I suspect it involved refinding such distances. I'm happy to have a look at restoring the `const` nature though - probably by holding distance between `Node` in the `Placement` base class at point of construction that is then queried.

Reply:
You could make the distance cache a `mutable` class member perhaps?